src/backend/distributed/shared_library_init.c

```diff
@@ -928,16 +950,38 @@ RegisterCitusConfigVariables(void)
 		GUC_STANDARD,
 		NULL, NULL, NULL);
 
+	DefineCustomIntVariable(
+		"citus.max_shared_pool_size",
+		gettext_noop("Sets the maximum number of connections allowed per worker node "
+					 "across all the backends from this node. Setting to -1 disables "
+					 "connections throttling. Setting to 0 makes it auto-adjust, meaning "
+					 "equal to max_connections."),
+		gettext_noop("For single coordinator setups, the value should be adjusted "
+					 "to match the max_connections on the remote nodes. For Citus MX, "
+					 "the value should be tuned such that it is roughly "
+					 "(max_connections / #workerNodes) so that each worker doesn't "
+					 "become overwhelmed by too many incoming connections"),
```

Comment:
Not sure how true the MX comment is, maybe omit until we study this in more detail.